### ISSUE TITLE
Enobio EEG and mouse clicks are offset for recordings longer than 5 minutes.

### ISSUE THREAD

**Issue by JVero** [sccn/labstreaminglayer#88]
We have been using LSL to record mouse clicks and Enobio32 EEG.  However, we have noticed some offsets in the data.  After starting the recording, we assume that the EEG should have the first time stamp, as we did not click for at least five seconds.  This proved to not be the case.  The first timestamp of the mouse buttons shows to be before the EEG's first time stamp, which should not be possible.  These are all on the same computer.  I have uploaded a sample data set with an empty EEG.  You can see the mouse clicks are timestamped to be way before the first timestamps of the EEG.

[Test_Blank_2_minutes.zip](https://github.com/sccn/labstreaminglayer/files/128766/Test_Blank_2_minutes.zip)

**Comment by dmedine** [comment]
I loaded this data in matlab via load_xdf:

`>> s = load_xdf('Test_Blank_2_minutes.xdf`

Then I printed out the meta data of the mouseclick stream and the EEG stream:

`>> s{2}.info % found the correct index by inspection'
`>> s{3}.info'

I see `first_timestamp: 4521.031520364' for the mouseclicks stream and `first_timestam: 4548.294419807' for the EEG stream. So, the first mouseclick is ~30 seconds ahead of the EEG stream. This is certainly much too long. 

I am not familiar with the Enobio device, but could it be an issue of having to have a particular window focused in order for the data to stream? That is just a shot in the dark, but an earlier version of the LeapMotion app had that issue.

**Comment by dmedine** [comment]
Another thing to try would be to use the LSL Enobio App. I understand you are using their native LSL implementation. It might be worth while to compare the two and see if they are similarly outrageously offset. 

I see that the last timestamp for the mouse is also well ahead of the last timestamp for the EEG. Can you verify if this is due to the way the data comes in or if there is a constant offset? Meaning, if you click
 several times just before stopping the recording, are those timestamps near to the last timestamps of the EEG or not?

**Comment by chkothe** [comment]
I just looked at the file, too. What must have happened is that the EEG
device only started sending data about 20 or so seconds after you started
the recording. In general it's not unusual that not all streams begin at
exactly the same time (even though they are still synchronized), because
the recording first needs to 'ramp' up for a second or two, but more than 5
seconds is quite unusual. What might be the case (although that's a guess)
is that the Enobio app perhaps checks if anyone is listening and only
starts the amp if that's the case. Most apps don't do that and just keep
the amp on all that time while the device is linked.

Christian

PS: To determine that I looked at the clock_times of that EEG stream, which
start around the same time as for the other streams.
